source/fundamentals/connection/tls.txt

```diff
@@ -0,0 +1,296 @@
+hostname verification, and setting the
+``tlsAllowInvalidCertificates`` to ``true`` disables certificate
+validation. Setting the ``tlsInsecure`` option to ``true`` disables
+both certificate and hostname validation.
+   
+.. warning::
+
+   Specifying any of these options in a production environment makes
+   your application insecure and potentially
+   vulnerable to expired certificates and to foreign processes posing
+   as valid client instances.
+
+For a full list of client options, see :ref:`node-connection-options`.
+
+.. _node-configure-tls-certificates:
+
+Configure Certificates
+----------------------
+
+To successfully initiate a TLS request, an application must prove its
+identity by referencing cryptographic certificates. To connect to
+MongoDB with TLS, your certificates must be stored as PEM
+files.
+
+.. important::
+
+   For production use, your MongoDB deployment should use valid
+   certificates generated and signed by the same certificate authority.
+   For testing, you can use self-signed certificates.
+
+The following list describes the components that you need to establish
+a connection with TLS:
+
+.. list-table::
+   :header-rows: 1
+   :widths: 30 70
+
+   * - TLS Component
+     - Description
+
+   * - Certificate Authority (CA)
+     - One or more certificate authorities to
+       trust when making a TLS connection.
+
+   * - Client Certificate
+     - A digital certificate and key that allow the server to verify the identity
+       of your application to establish an encrypted network connection.
+
+   * - Certificate Key
+     - The client certificate private key file. This key is often
+       included within the certificate file itself.
+
+   * - Passphrase
+     - The password to decrypt the client private key if it is encrypted.
+
+.. tip::
+   
+   To learn more about the PEM format, see the Wikipedia entry on
+   :wikipedia:`Privacy-Enhanced Mail <Privacy-Enhanced_Mail>`.
+
+.. _node-client-tls-connect:
+
+Reference Certificates in a Client
+----------------------------------
+
+You must reference your certificates in your ``MongoClientOptions``
+object so that the server can validate them before the client connects.
+You can reference your certificates in the following ways:
+
+- Create a ``SecureContext`` object to store certificates *(Recommended)*
+- Provide filepath strings that point to your certificates
+- Create ``Buffer`` objects to store certificates
+
+.. _node-tls-securecontext:
+
+Create a ``SecureContext`` Object to Store Certificates
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+We recommend that you use the ``secureContext`` option to configure
+your TLS connection. ``SecureContext`` objects are native to Node.js
+and allow you to keep all your TLS options in a single reusable object.
+
+To create a ``SecureContext`` object, you must first install the ``tls``
+package and use it in your application. Next, call the ``tls.createSecureContext()``
```

Comment:
Should be:

> To create a ``SecureContext`` object, import the ``createSecureContext`` function from the ``tls`` module.

The `tls` module is a standard library package available on all versions of Node.js